src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprRecipeGroup.java

```diff
@@ -47,36 +44,32 @@ public class ExprRecipeGroup extends PropertyExpression<Recipe, String> {
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (exprs[0].isDefault() && getParser().isCurrentEvent(CreateRecipeEvent.class))
 			isEvent = true;
-		//noinspection unchecked
-		setExpr((Expression<? extends Recipe>) exprs[0]);
-		return true;
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
 	}
 
 	@Override
-	protected String[] get(Event event, Recipe[] source) {
-		return get(source, recipe -> {
-			if (recipe instanceof MutableRecipe)  {
-				if (recipe instanceof MutableCraftingRecipe mutableCraftingRecipe) {
-					return mutableCraftingRecipe.getGroup();
-				} else if (recipe instanceof MutableCookingRecipe mutableCookingRecipe) {
-					return mutableCookingRecipe.getGroup();
-				} else if (recipe instanceof MutableStonecuttingRecipe mutableStonecuttingRecipe) {
-					return mutableStonecuttingRecipe.getGroup();
-				}
-			} else {
-				// TODO: Combine ShapedRecipe and ShapelessRecipe into CraftingRecipe when minimum version is raised to 1.20.1 or higher.
-				if (recipe instanceof ShapedRecipe shapedRecipe) {
-					return shapedRecipe.getGroup();
-				} else if (recipe instanceof ShapelessRecipe shapelessRecipe) {
-					return shapelessRecipe.getGroup();
-				} else if (recipe instanceof CookingRecipe<?> cookingRecipe) {
-					return cookingRecipe.getGroup();
-				} else if (recipe instanceof StonecuttingRecipe stonecuttingRecipe) {
-					return stonecuttingRecipe.getGroup();
-				}
+	public @Nullable String convert(Recipe recipe) {
+		if (recipe instanceof MutableRecipe)  {
+			if (recipe instanceof MutableCraftingRecipe mutableCraftingRecipe) {
+				return mutableCraftingRecipe.getGroup();
+			} else if (recipe instanceof MutableCookingRecipe mutableCookingRecipe) {
+				return mutableCookingRecipe.getGroup();
+			} else if (recipe instanceof MutableStonecuttingRecipe mutableStonecuttingRecipe) {
+				return mutableStonecuttingRecipe.getGroup();
```

Comment:
we could make a Grouped interface for the mutable ones at least
i just really don't like these massive ifelses